### DISCUSSION TITLE
Cops devs, we need a "rubular.com" for creating and debugging S-expressions

### DISCUSSION THREAD

**Discussion by skull-squadron** [rubocop/rubocop#11746] [Ideas]
I'm constantly debugging S-expressions with rspec, which doesn't help all that much. There has to be an easier way.

### Requirements

a. User enters the name of the cop they want to create `Whatever/Foo/Hey` , and a new default template appears
b. Live reevaluation whenever text in any panel changes

### Design

1. Cop panel with a skeleton cop class
2. Config panel with a blank but editable rubocop.yml
3. Input panel similar to a gist with an ability to create multiple files but with ad-hoc subdirectories `foo/bar/hey.rb`
4. Output panel with a simulated rubocop run
5. Errors panel if something is invalid
6. Guide panel with S-expression syntax and links to doc

### Stretch goals

A. Context-sensitive S-expression completion / help
B. Errors show in the appropriate panel and remove the errors panel

### Non-requirements

I. "Fiddle" / playground permalink not necessary

### Notes

i. Perhaps Opal to run it in the browser similar to TryRuby

**Comment by bbatsov** [comment]
Have you seen https://nodepattern.herokuapp.com/ ?

**Comment by dvandersluis** [comment]
There's https://nodepattern.herokuapp.com/ which @marcandre created.

Also something I find useful is running `ruby-parse -e "[your code]"` from the command line which will output the s-expression for the code.
